### PR TITLE
Fix missing monster spell casting during battles

### LINKED ISSUE #3821: Monsters don't cast spells anymore
For instance, Mummies should be able to cast curse randomly.
This doesn't seem to be the case anymore

### PR DESCRIPTION
**Pull request by ihhub**
close #3821

### CODE REVIEW

File: src/fheroes2/monster/monster_info.cpp [hunk 1]
```diff
@@ -134,78 +134,78 @@ namespace
...
-            { 8, 6, 4, 6, 40, Speed::FAST, 0, std::set<fheroes2::MonsterAbility>(), std::set<fheroes2::MonsterWeakness>() }, // Fire Elemental
-            { 6, 8, 3, 7, 45, Speed::AVERAGE, 0, std::set<fheroes2::MonsterAbility>(), std::set<fheroes2::MonsterWeakness>() }, // Water Elemental
-            { 0, 0, 0, 0, 0, Speed::VERYSLOW, 0, std::set<fheroes2::MonsterAbility>(), std::set<fheroes2::MonsterWeakness>() }, // Random Monster
-            { 0, 0, 0, 0, 0, Speed::VERYSLOW, 0, std::set<fheroes2::MonsterAbility>(), std::set<fheroes2::MonsterWeakness>() }, // Random Monster 1
-            { 0, 0, 0, 0, 0, Speed::VERYSLOW, 0, std::set<fheroes2::MonsterAbility>(), std::set<fheroes2::MonsterWeakness>() }, // Random Monster 2
-            { 0, 0, 0, 0, 0, Speed::VERYSLOW, 0, std::set<fheroes2::MonsterAbility>(), std::set<fheroes2::MonsterWeakness>() }, // Random Monster 3
-            { 0, 0, 0, 0, 0, Speed::VERYSLOW, 0, std::set<fheroes2::MonsterAbility>(), std::set<fheroes2::MonsterWeakness>() }, // Random Monster 4
+            { 0, 0, 0, 0, 0, Speed::VERYSLOW, 0, std::vector<fheroes2::MonsterAbility>(), std::vector<fheroes2::MonsterWeakness>() }, // Unknown Monster
```
Comment (qarkai) [inline src/fheroes2/monster/monster_info.cpp:137]: Is it possible to use simple `{}` instead of `std::vector<...>()`?

Comment (ihhub) [reply src/fheroes2/monster/monster_info.cpp:138]: Good catch! I'll do it.

**Comment by ihhub** [comment]
Hi @oleg-derevenetz and @vincent-grosbois  as proposed previously there are changes to fix missing monster spell casting. Of course it's not an ideal solution for Monsters and for Artifact (design wise) but at least it's better than before.

File: src/fheroes2/monster/monster_info.cpp [hunk 2]
```diff
@@ -757,66 +765,66 @@ namespace fheroes2
             }
         }
 
-        foundAbility = abilities.find( MonsterAbility( MonsterAbilityType::IMMUNE_TO_CERTAIN_SPELL, 0, spellId ) );
+        foundAbility = std::find( abilities.begin(), abilities.end(), MonsterAbility( MonsterAbilityType::IMMUNE_TO_CERTAIN_SPELL, 0, spellId ) );
         if ( foundAbility != abilities.end() ) {
```
Comment (oleg-derevenetz) [inline src/fheroes2/monster/monster_info.cpp:769]: Shouldn't we additionally verify the `spellId` here?

Comment (ihhub) [reply src/fheroes2/monster/monster_info.cpp:769]: We can do this at the beginning of the function.

Comment (ihhub) [reply src/fheroes2/monster/monster_info.cpp:769]: Ah, yes. You're right.

**Review by oleg-derevenetz — APPROVED** [review]

**Pull request merged**